war/models.py

```diff
@@ -72,17 +72,26 @@ def bin_max(value):
         return value
 
 
-def survivor_signature(value):
-    machine_code = value.file.read()
-    if not machine_code:
-        raise ValidationError("Survivor may not be empty.")
+class SurvivorSignatureValidator:
```

Comment:
This class validates more than the signature (non-empty survivors), maybe rename it to something along the lines of `SurvivorCodeValidator`